routes/deviceReg.js

```diff
@@ -0,0 +1,32 @@
+var express = require('express');
+var router = express.Router();
+var Choral = require('../models/choral');
+
+router.get('/', function(req, res, next) {
+  res.render('deviceReg', {
+    googleUser: req.user });
+});
+
+router.post('/', function(req, res, next) {  
+  var attrs = {
+    user: res.locals.userModel,
+    sampleRate: req.body.sampleRate,
+    type: 'device',
+    name: req.body.deviceName
+  };
+
+  var device = Choral.createNew(attrs, (err, c) => {
```

Comment:
instead of doing this, do 

```
Choral.createNew(attrs, (err, device) => {
  // database query is guaranteed to be finished inside this callback
  if(err){
     console.log(err);
     return next(err);
   }
   // Device is guaranteed to have been successfully created if we reach this line.
   res.render('deviceReg', {googleUser: req.user, devID: device.choralId}); // use the device here
 }); 
```